pensieve/config.py

```diff
@@ -163,18 +228,30 @@ class MetricDefinition:
     name: str  # implicit in configuration
     select_expression: str
     data_source: DataSourceReference
+    statistics: Dict[str, Dict[str, Any]]
 
-    def resolve(self, spec: "AnalysisSpec") -> mozanalysis.metrics.Metric:
+    def resolve(self, spec: "AnalysisSpec") -> List[Summary]:
         select_expression = _metrics_environment.from_string(self.select_expression).render()
 
-        return mozanalysis.metrics.Metric(
+        metric = mozanalysis.metrics.Metric(
             name=self.name,
             data_source=self.data_source.resolve(spec),
             select_expr=select_expression,
         )
 
+        metrics_with_treatments = []
+
+        for statistic_name, params in self.statistics.items():
+            for statistic in Statistic.__subclasses__():
+                if statistic.name() == statistic_name:
+                    metrics_with_treatments.append(Summary(metric, statistic.from_dict(params)))
+                else:
+                    raise ValueError(f"Statistic {statistic_name} does not exist.")
+
+        return metrics_with_treatments
```

Comment:
I think this should error instead of returning an empty list if there are no Statistics.